core/src/main/java/org/elasticsearch/search/aggregations/metrics/percentiles/PercentilesAggregationBuilder.java

```diff
@@ -293,4 +313,29 @@ protected int innerHashCode() {
     public String getType() {
         return NAME;
     }
+
+    /**
+     * Private specialization of this builder that should only be used by the parser, this enables us to
+     * overwrite {@link #method()} to check that it is not called twice when parsing xContent and throw
+     * an error while the Java API should allow this
+     */
+    private static class InternalBuilder extends PercentilesAggregationBuilder {
```

Comment:
In a first wip of ths PR I was doing the checks if the setter was called multiple times in the original builder, but that felt akward since it would also affect the java api and would mean the PercentilesAggregationBuilder would somehow need that call counter somewhere. Attempts to use ConstructingObjectParser to check that methods was specified only once also were not promosing, so I went with this internal temporary object for the parser output for now as it felt least invasive.